jwst/outlier_detection/outlier_detection_ifu.py

```diff
@@ -53,203 +49,258 @@ def __init__(self, input_models, reffiles=None, **pars):
-            # channels (MIRI) of data into a single frame to match the
-            # original input...
-            self.blot_median(median_model)
+        # How many unique values of detector?
+        uq_det = np.unique(detector)
+        ndet = len(uq_det)
+        for idet in range(ndet):
+            indx = (np.where(detector == uq_det[idet]))[0]
+            ndet_files = int(len(indx))
+            self.flag_outliers(idet, uq_det, ndet_files,
+                               diffaxis, nx, ny,
+                               kern_size, threshold_percent,
+                               save_intermediate_results,
+                               ifu_second_check)
+
+        # send input_models back to outlier_detection.py.
+        # self.input_moels is  that is what is returned from outlier_detection.py
+        self.detect_outliers_ifu(self.input_models)
+
+    def flag_outliers(self, idet, uq_det, ndet_files,
+                      diffaxis, nx, ny,
+                      kern_size, threshold_percent,
+                      save_intermediate_results,
+                      ifu_second_check):
+        """
+        Flag outlier pixels on IFU. In general we are searching for pixels that
+        are a form of a bad pixel but not in bad pixel mask, because the bad pixels vary with
+        time. This program will flag the DQ of input images as DO_NOT_USE and OUTLIER and set
+        the associated science pixel to a Nan. This routine only works on data from one detector. 
+
+        Parameters
+        ----------
+        idet : int
+            Integer indicating which detector we are working with
+        uq_det : string array
+            Array of (unique) detector names found input data 
+        n_det_files : int
+            Number of files for the detector we are working on 
+        diffaxis : int
+            The axis to form the adjacent pixel differences
+        nx : int
+            Size of input data on x axis
+        ny : int
+            Since of inut data on y axis
+        threshold_percent : float
+            Percent for flagging outliers. Flags pixels where the minimum difference between
+            adjacent pixels for all the input data for a detector is above this percentage. The
+            percentage is based on using all the pixels except a 4 X 4 row and column region around
+            the detector that is often noisy.
+        save_intermediate_results : boolean
+            If True then save intermediate output data
+        ifu_second_check : boolean
+            If True then perform a secondary check searching for outliers. This will set outliers
+            whereever the difference array of adjacent pixels is a Nan. 
+        """
+
+        # set up array to hold group differences
+        diffarr = np.zeros([ndet_files, ny, nx])
+        j = 0
+        for i, model in enumerate(self.input_models):
+            sci = model.data
+            dq = model.dq
```

Comment:
Since `model.data` and `model.dq` are worked with only when the detector matches, these can go in the `if` block.